### ISSUE TITLE
Instance for M1 causes LH to crash

### ISSUE THREAD

**Issue by davidspies** [ucsd-progsys/liquidhaskell#1941]
Here's a minimal change to the LH Plugin Demo demonstrating it:

https://github.com/davidspies/lh-plugin-demo/commit/a4b538210cfe53c96fc8eea6efbbab657a27b3a6

When no `.liquid` folder exists already, running `stack build` results in the error:

```
src/Demo/.liquid: copyFile:atomicCopyFileContents:withReplacementFile:openBinaryTempFile: does not exist (No such file or directory)
```

Strangely, this only happens if there's an instance for `M1`; not for any of the other Generic wrappers.

**Comment by ranjitjhala** [comment]
Yikes, and this is just the plain ghc-8.10.2 also...

Just so I follow, if you create a folder `src/Demo/.liquid` does it work?
[Don't have access to an M1 machine yet, unfortunately...]

**Comment by davidspies** [comment]
When I do that, I get

```
**** LIQUID: ERROR src/Demo/Lib.hs:15:3-15:30: Error
  elaborate solver failed on:
      papp2 q VV##F##38 zcmp
  with error
      Unbound symbol q --- perhaps you meant: len, snd, fst ?
  in environment
      VV##F##38 := (f##a19U p##a1a1)

      papp2 := func(4 , [(Pred @(0) @(1)); @(2); @(3); bool])

      zcmp := (f##a19U p##a1a1)
```

**Comment by nikivazou** [comment]
By `M1` you mean Apple's 2020 chip? I am using this and have no troubles....

**Comment by davidspies** [comment]
No, I mean the datatype https://hackage.haskell.org/package/base-4.16.0.0/docs/GHC-Generics.html#t:M1
Presumably the issue is OS-indepedent
(I'm on Ubuntu)

**Comment by ThreeEyedGod** [comment]
Hi was there a resolution to this problem? Facing identical issue
